tools/preprocess/README.md

```diff
@@ -142,3 +142,27 @@ python tools/preprocess/reformat_jsonl_nan_value.py --help
 - `src_dir`: you just need to set this argument to the path which stores filenames are like `*.jsonl`.
 - `target_dir`: result directory to store the converted jsonl files.
 - `num_proc` (optional): number of process workers. Default it's 1.
+
+
+### Serialize meta fields in jsonl file
+
+In some jsonl files, different samples may have different meta fields, and even the data types in the same meta field may be different, which will cause failure to read the dataset using [HuggingFace Dataset](https://huggingface.co/docs/datasets/index). This tool is used to serialize all meta fields except `text_key` in these jsonl files into strings to facilitate subsequent Data-juicer processing. After the dataset is processed, it usually needs to be deserialized using [deserialize_meta.py](../postprocess/deserialize_meta.py).
+
+
+
+```shell
+python tools/preprocess/serialize_meta.py           \
+    --src_dir           <src_dir>         \
+    --target_dir        <target_dir>      \
+    --text_key          <text_key>        \
```

Comment:
This tool might need to support multiple text_keys to align with the argument `text_keys_to_load` in dj.